test/gtest/typecheck_test.cpp

```diff
@@ -0,0 +1,62 @@
+/*
```

Comment:
For unit tests, we want as much control as possible. I think we're better off using an approach like this. It gives us maximum flexibility, and it has the advantage of not invoking `cpp` for every unit test.